src/renderer/screens/exchange/Swap2/Form/FormSelectors/FormInputs.js

```diff
@@ -20,12 +31,57 @@ function SwapButton() {
   );
 }
 
+export type toAccountType =
+  | {
+      // User already has an account to accept target currency
+      targetAccountExists: true,
+      data: {
+        account: ?Account,
+        subAccount: ?SubAccount,
+        currency: ?(TokenCurrency | CryptoCurrency),
+      },
+    }
+  | {
+      // User doesn't have an account to accept target currency
+      targetAccountExists: false,
+      data: {
+        account: null,
+        subAccount: null,
+        currency: ?(TokenCurrency | CryptoCurrency),
+      },
+    }
+  // No target currency selected
+  | null;
+
 export default function FormInputs() {
-  const [fromAccount, setFromAccount] = useState(null);
+  // TODO: would be moved to a reducer
+  const [fromAccount, setFromAccount] = useState<SwappableAccountType>(null);
   const [fromAmount, setFromAmount] = useState(null);
-  const [toCurrency, setToCurrency] = useState(null);
+  const [toAccount, setToAccount] = useState<toAccountType>(null);
   const [toAmount, setToAmount] = useState(null);
 
+  // TODO: would be handled by the reducer in the future
+  const handleSetToAccountChange = (selectSate: useSelectableCurrenciesReturnType) => {
```

Comment:
There's a reducer in logic.js in live-common for swap that does just this, perhaps you can rework it, or redo it, or something.

Reply:
Yeah definitely. I plan to use/modify it once I'll start to work on the reducer logic (moving states one-step higher, from FromInput to the form container)

Reply:
👍🏼